collect_app/src/main/java/org/odk/collect/android/injection/config/AppDependencyModule.java

```diff
@@ -185,9 +187,18 @@ UserAgentProvider providesUserAgent() {
 
     @Provides
     @Singleton
-    public OpenRosaHttpInterface provideHttpInterface(MimeTypeMap mimeTypeMap, UserAgentProvider userAgentProvider) {
+    public OpenRosaHttpInterface provideHttpInterface(MimeTypeMap mimeTypeMap, UserAgentProvider userAgentProvider, StoragePathProvider storagePathProvider, SettingsProvider settingsProvider) {
+        Settings settings = settingsProvider.getUnprotectedSettings();
+
+        String cacheDir;
+        if (settings.getBoolean(ProjectKeys.KEY_EXPERIMENTAL_HTTP_CACHING)) {
+            cacheDir = storagePathProvider.getOdkDirPath(StorageSubdirectory.CACHE);
```

Comment:
I had a realization that using the project's cache dir doesn't really make sense: you'd end up with the cache for every project in the cache dir of the current project when the first request of the process is made.

I think we either need to have the cache in our private cache dir, or retain multiple OkHttp client with one for each project. I'm a little concerned that the latter is something we should be doing anyway, so I'll mark this as a draft and investigate.